extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,145 @@
+                url=XsUri(node.conanfile.homepage),
+            ))
+        return component
+
+    def me_as_tool() -> Tool:
+        tool = Tool(name="conan extension recipe:create-sbom")
+        tool.external_references.add(ExternalReference(
+            type=ExternalReferenceType.WEBSITE,
+            url=XsUri("https://github.com/conan-io/conan-extensions")))
+        return tool
+
+    formatters = {
+        '1.4_json': cyclonedx.output.json.JsonV1Dot4,
+        '1.3_json': cyclonedx.output.json.JsonV1Dot4,
+        '1.2_json': cyclonedx.output.json.JsonV1Dot4,
+        '1.4_xml': cyclonedx.output.xml.XmlV1Dot4,
+        '1.3_xml': cyclonedx.output.xml.XmlV1Dot3,
+        '1.2_xml': cyclonedx.output.xml.XmlV1Dot2,
+        '1.1_xml': cyclonedx.output.xml.XmlV1Dot1,
+        '1.0_xml': cyclonedx.output.xml.XmlV1Dot0
+    }
+    parser.add_argument("--sbom_format", choices=formatters.keys(), required=True)
+
+    # region COPY FROM conan: cli/commands/graph.py
+    common_graph_args(parser)
+    args = parser.parse_args(*args)
+    validate_common_graph_args(args)
+    cwd = os.getcwd()
+    path = conan_api.local.get_conanfile_path(args.path, cwd, py=None) if args.path else None
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile,
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update)
+    # endregion COPY
+
+    components = {node: create_component(node) for node in deps_graph.nodes}
+    bom = Bom()
+    bom.metadata.component = components[deps_graph.root]
+    bom.metadata.tools.add(me_as_tool())
+    for node in deps_graph.nodes[1:]:  # node 0 is the root
+        bom.components.add(components[node])
+    for dep in deps_graph.nodes:
+        bom.register_dependency(components[dep], [components[dep_dep.dst] for dep_dep in dep.dependencies])
+    return {'bom': bom, 'sbom_format': args.sbom_format, 'formatters': formatters}
```

Comment:
Are the different formats working? I might be doing something wrong, but
```
conan recipe:create-sbom --sbom_format 1.3_json .
```
results for me in
```
[...] "specVersion": "1.4"}
```

Reply:
I noticed that the xml versions are working; is this an issue in cyclonedx-python-lib?

Reply:
Copy and paste error. I will fix it

Reply:
Should have seen that myself as well...

Reply:
Fixed